### DISCUSSION TITLE
Where or what file put this configuration?

### DISCUSSION THREAD

**Discussion by kevincharp** [JanDeDobbeleer/oh-my-posh#5120] [Troubleshoot]
I didn't understand where you have to put this configuration so that it updates automatically

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/85292141/a6fce30d-30b2-4a9f-b3ac-98240637b62a)

**Comment by lewis-yeung** [comment]
Just add it in your config (theme) file, e.g., in JSON format: `"auto_upgrade": true`. The description for this setting is missing in the documentation. I've made a PR to add it.

**Reply by kevincharp** [reply]
I understand that I should wait for the new version change to be able to configure the auto upgrade, is that correct?

I still don't see the lines modified in the last PR reflected in my scheme json file

**Reply by JanDeDobbeleer** [reply]
@kevincharp no, you can already add this.

**Reply by kevincharp** [reply]
Thanks! I'm going to try it

**Reply by kevincharp** [reply]
I have a slight feeling that when I do update/winget upgrade oh-my-posh the scheme.json file is rewritten and the auto_upgrade setting changes back to its default value false. The message suggesting that I make this configuration appears again.

Am I doing something wrong?

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/85292141/fafb22c3-97d7-44bd-bd4e-2065101b9e5e)

**Reply by JanDeDobbeleer** [reply]
@kevincharp if you want to use and customize one of the theme configurations, copy paste it elsewhere as the installer will indeed update (and thus replace) these when upgrading.

**Reply by bombenbodo** [reply]
I just mirrored and renamed the monty Theme but kept it in the default theme directory and it hasn't been overwritten since.
So, to add auto_upgrade and disable update notice  to my custom theme I simply added both entries to the beginning of my 'monty.custom.omp.json'
and it worked.
e.g.
>{
"$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
"auto_upgrade": true,
"disable_notice": true,
"blocks": [
 .....

Thx

**Reply by DGWIIImelt** [reply]
> monty.custom.omp.js

is that the expected configuration to allow autoupgrade? seems a bit a sketch.

**Reply by JanDeDobbeleer** [reply]
@DGWIIImelt if you want to enable auto upgrade and not have notifications other than actual upgrades, this is what needs to be added:

```json
"auto_upgrade": true,
"disable_notice": true,
```

**Comment by YovelB** [comment]
Thanks. it makes it a lot easier to update.
Here's how to do it in a toml file if anyone is wondering.

![Screenshot from 2024-08-15 21-08-57](https://github.com/user-attachments/assets/e39ca02a-f779-4005-a5fd-34ad7fa95ba2)
